### DISCUSSION TITLE
Support RPMSCRIPT_FLAG_CRITICAL for posttrans

### DISCUSSION THREAD

**Discussion by lclgo** [rpm-software-management/rpm#3044] [General]
**Is your feature request related to a problem? Please describe.**

When upgrading kernel, we will make initrd in `posttrans` script. But the process of making initrd may fail in some unusual situations, and we only get a warning message, the rpm command returns 0. 

If the rpm command is running in a script, we will treat it as a successful kernel update by mistake, and then reboot. But becasuse the initrd wasn't successfully built, the reboot will enter emergency mode.

**Describe the solution you'd like**

Support some options to make `posttrans` script (can also be used in `post`, `postun`...) failure as RPMSCRIPT_FLAG_CRITICAL.

**Describe alternatives you've considered**

Add a new `posttrans_critical`, it supplies the same function as `posttrans` except the error of script is treated as critical.

**Comment by pmatilai** [comment]
The current behavior is a mistake really, and one that we'll be reverting in 6.0: https://github.com/rpm-software-management/rpm/issues/2581

**Reply by lclgo** [reply]
Thanks for your response. So can I just revert the related patches introduced in 4.6.0 for now? I'm worry that you might change some behaviors in 6.0 and it will be somewhat different from 4.6.0.
